### PR TITLE
*: support create table with vector indexes and others | tidb-test=c567cde7e19a55a67960a404906cb859ca44f206

### PR DESCRIPTION
**Pull request by zimulala**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/56122

Problem Summary:
* Support for adding/dropping vector index during table creation in the DDL
  * If the table hasn't TiFlash replica, default set it count to 1.
* Unsupported operations:
  * admin cleanup index
  * set the vector index to invisible

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/meta/model/bdr.go [hunk 1]
```diff
@@ -101,6 +101,7 @@ var BDRActionMap = map[DDLBDRType][]ActionType{
 		ActionReorganizePartition,
 		ActionAlterTablePartitioning,
 		ActionRemovePartitioning,
+		ActionAddVectorIndex,
```
Comment (tangenta) [inline pkg/meta/model/bdr.go:104]: Could you please ask TiCDC developers to confirm this?

Comment (zimulala) [reply pkg/meta/model/bdr.go:104]: @asddongmen PTAL

Comment (asddongmen) [reply pkg/meta/model/bdr.go:104]: After being classified as unsafe DDL, TiCDC won't synchronize this DDL in BDR mode. So, it won't affect the original BDR functionality.

File: pkg/ddl/column.go [hunk 2]
```diff
@@ -1094,10 +1094,10 @@ func isColumnWithIndex(colName string, indices []*model.IndexInfo) bool {
 
 func isColumnCanDropWithIndex(colName string, indices []*model.IndexInfo) error {
 	for _, indexInfo := range indices {
-		if indexInfo.Primary || len(indexInfo.Columns) > 1 {
+		if indexInfo.Primary || len(indexInfo.Columns) > 1 || indexInfo.VectorInfo != nil {
 			for _, col := range indexInfo.Columns {
 				if col.Name.L == colName {
-					return dbterror.ErrCantDropColWithIndex.GenWithStack("can't drop column %s with composite index covered or Primary Key covered now", colName)
+					return dbterror.ErrCantDropColWithIndex.GenWithStack("can't drop column %s with composite index covered or Primary Key or Vector Key covered now", colName)
```
Comment (tangenta) [inline pkg/ddl/column.go:1100]: Can we split these messages for each case?

File: pkg/ddl/index.go [hunk 3]
```diff
@@ -879,11 +884,12 @@ func (w *worker) checkVectorIndexProcessOnTiFlash(jobCtx *jobContext, t *meta.Me
 
 func (w *worker) checkVectorIndexProcess(jobCtx *jobContext, tbl table.Table, job *model.Job, index *model.IndexInfo) error {
 	waitTimeout := ReorgWaitTimeout
+	ticker := time.Tick(waitTimeout)
```
Comment (tangenta) [inline pkg/ddl/index.go:887]: Use `time.NewTick()` instead to prevent resource leak.

**Review by tangenta — COMMENTED** [review]
Rest LGTM

**Review by tangenta — APPROVED** [review]

File: pkg/ddl/create_table.go [hunk 4]
```diff
@@ -509,6 +509,40 @@ func checkGeneratedColumn(ctx sessionctx.Context, schemaName pmodel.CIStr, table
...
 
+func checkVectorIndexIfNeedTiFlashReplica(ctx sessionctx.Context, tblInfo *model.TableInfo) error {
+	var hasVectorIndex bool
+	for _, idx := range tblInfo.Indices {
+		if idx.VectorInfo != nil {
+			hasVectorIndex = true
+			break
+		}
+	}
+
```
Comment (wjhuang2016) [inline pkg/ddl/create_table.go:520]: if !hasVectorIndex {
return nil
}

**Review by wjhuang2016 — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Pull request merged**